### ISSUE TITLE
Can't update extension (missing version number)

### ISSUE THREAD

**Issue by hho** [Azure/azure-iot-cli-extension#67]
When trying to update the extension, the following error is thrown:

```
$ az extension update --name azure-cli-iot-ext
'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 328, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 386, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 379, in _run_job
    six.reraise(*sys.exc_info())
  File "/opt/az/lib/python3.6/site-packages/six.py", line 693, in reraise
    raise value
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 356, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 171, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 452, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/extension/operations.py", line 259, in update_extension
    cur_version = ext.get_version()
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/extension/__init__.py", line 108, in get_version
    return self.metadata.get('version')
AttributeError: 'NoneType' object has no attribute 'get'
```

And sure enough, the extension doesn't seem to have a version number:
```
$ az extension list
[
  {
    "extensionType": "whl",
    "name": "azure-cli-iot-ext",
    "version": null
  }
]
```

Digging further I found out, that this seems to be due to a change in `wheel`: https://github.com/Azure/azure-cli/issues/6441

Please package a version with the old version of `wheel`. It seems users using 0.6.1 can't update from there anymore (will have to remove&add when the new extension version is out).

**Comment by c-ryan-k** [comment]
Thanks for catching this @hho - I've updated the release artifact with one built with wheel 0.30.0.

**Comment by digimaun** [comment]
FYI v0.6.1 uses v0.30.0 of the wheel package which is the part of the compatible set listed in the linked issue (0.29.0 or 0.30.0).

**Comment by hho** [comment]
@digimaun Interesting. So my first diagnosis seems to be wrong (I checked, there is a `metadata.json` in the whl file) – but the problem persists: No version, no update. :disappointed:

**Comment by digimaun** [comment]
I was easily able to repro the issue with CLI version 2.0.55. Then upgraded to latest v2.0.60 which had the same issue (as you already discovered).

I also tested this issue on an older version of Az CLI (version 2.0.52) installed in a different environment. That _correctly_ shows the extension version.

So some change in CLI core between version 2.0.53 and 2.0.55 caused the issue which still persists today in the latest CLI release.

**Comment by hho** [comment]
For fun, I just installed _all_ available extensions…
There are only **3** who don't report a version number with CLI 2.0.60:
* mesh
* azure-batch-cli-extensions
* azure-cli-iot-ext

So it somehow does depend on both, CLI version _and_ the extension itself. :thinking:

**Comment by jongio** [comment]
I ran into this today.  What is the workaround?

**Comment by jongio** [comment]
I removed and readded and that seemed to be a good workaround


Remove: `az extension remove --name azure-cli-iot-ext`


Add: `az extension add --name azure-cli-iot-ext`

**Comment by anusapan** [comment]
This issue was related to CLI Core and it has been fixed into CLI Core. The impact will come into the act from the next release.

**Comment by digimaun** [comment]
@hho After updating can you please confirm the issue is resolved for you? I'm using CLI version `2.0.62` and the correct IoT extension metadata is shown in my environment.

**Comment by hho** [comment]
I can confirm that the issue is gone with CLI 2.0.62. :tada: 

Also, `az extension update` now seems to work correctly. :raised_hands:
